storagemarket/impl/client.go

```diff
@@ -65,11 +75,22 @@ type clientDealUpdate struct {
 	mut      func(*ClientDeal)
 }
 
-func NewClient(h host.Host, dag ipld.DAGService, dataTransfer datatransfer.Manager, discovery *discovery.Local, deals *statestore.StateStore, scn storagemarket.StorageClientNode) *Client {
+func NewClient(h host.Host, bs blockstore.Blockstore, dataTransfer datatransfer.Manager, discovery *discovery.Local, deals *statestore.StateStore, scn storagemarket.StorageClientNode) (*Client, error) {
+	pr := padreader.NewPadReader()
+	carIO := cario.NewCarIO()
+	sectorCalculator := sectorcalculator.NewSectorCalculator("")
```

Comment:
non-blocking:
I think when we move to integration we may want implementations to setup PieceIO themselves and pass it as a depedency, along with the filestore.

I continue to think we should have pieceIO have a blockstore as well so we don't need it as a depedency here.